### PR TITLE
Allow extra post deploy update tasks

### PR DESCRIPTION
**Pull request by kentr**
How about something like this for #34?

I've tested it with the `drush` commands defined in `main.yml`.

The `cache clear` command could be defined via a conditional based on `drupal_major_version`, something like

    - command: "{{ drush_path }} {{ (drupal_major_version >= 8) | ternary('cache-rebuild','cc all') }} --quiet"
      chdir: "{{ drupal_core_path }}"

### CODE REVIEW

**Comment by kentr** [comment]
Hmm.  I might be conflating the role with DrupalVM.  

Is `drupal_major_version` not used in this role?

Either way, could be detected programmatically with a `drush` call.

File: tasks/update.yml [hunk 1]
```diff
@@ -1,13 +1,22 @@
 ---
-- name: Run database updates.
-  command: "{{ drush_path }} updatedb -y"
+
+- name: Detect Drupal version.
+  command: drush status drupal-version --format=list
```
Comment (oxyc) [inline tasks/update.yml]: This should use `{{ drush_path }}` as well

File: tasks/update.yml [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
-# TODO: Other commands if configured?
+- name: Run post-update tasks.
+  command: "{{ item.command }}"
+  args:
+    chdir: "{{ item.chdir | default(omit) }}"
+  with_items: "{{ drupal_deploy_update_tasks }}"
 
 - name: Rebuild Drupal caches.
-  command: "{{ drush_path }} cache-rebuild --quiet"
+  command: "{{ drush_path }} {{ (drupal_core_version >= '8') | ternary('cache-rebuild','cc all') }} --quiet"
```
Comment (oxyc) [inline tasks/update.yml]: Guess the comparison should always work (?) but there's also the safer [`version_compare`](http://docs.ansible.com/ansible/latest/playbooks_tests.html#version-comparison) filter.

File: tasks/update.yml [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+- name: Set Drupal version.
+  set_fact:
+    drupal_core_version: "{{ drupal_core_version_output.stdout }}"
 
-# TODO: Import configuration if configured?
-# TODO: Other commands if configured?
+- name: Run post-update tasks.
+  command: "{{ item.command }}"
+  args:
+    chdir: "{{ item.chdir | default(omit) }}"
```
Comment (oxyc) [inline tasks/update.yml]: Personally I think `chdir` could be hardcoded to `drupal_core_path` without an option to change. I doubt anyone would ever needs to run anything from the home directory (not even sure if it's root or vagrant).

Comment (kentr) [inline tasks/update.yml]: I've encountered use-cases for the path not being `drupal_core_path`, and specifying it as a command option vs `chdir` seems partly a style preference to me.  Allowing override here also prevents accidentally excluding some command that doesn't have a path option.  I can't think of one off the top of my head, but that doesn't mean there isn't one...

I can also see the convenience of defaulting to `drupal_core_path`.

What about allowing override, but with a default to `drupal_core_path`?

File: tasks/update.yml (hunk 2 above)
Comment (kentr) [inline tasks/update.yml]: Yeah, that's definitely safer.  Thanks for pointing it out.

File: tasks/update.yml (hunk 3 above)
Comment (kentr) [reply tasks/update.yml]: @oxyc ^^

Comment (kentr) [reply tasks/update.yml]: @oxyc @geerlingguy 

Any chance of getting this merged after the conflicts are resolved?

Regarding `chdir`: The current code uses `chdir` anyway, so the PR is consistent with the current code in that regard.

File: tasks/update.yml (hunk 1 above)
Comment (kentr) [reply tasks/update.yml]: This was resolved in 0185c10

**Pull request closed without merging**